sklearn/metrics/pairwise.py

```diff
@@ -1537,14 +1537,16 @@ def sigmoid_kernel(X, Y=None, gamma=None, coef0=1):
     array([[0.76..., 0.76...],
            [0.87..., 0.93...]])
     """
+    xp, _ = get_namespace(X, Y)
     X, Y = check_pairwise_arrays(X, Y)
     if gamma is None:
         gamma = 1.0 / X.shape[1]
 
     K = safe_sparse_dot(X, Y.T, dense_output=True)
```

Comment:
I'm confused, `safe_sparse_dot` seems to _support_ array API, but at the end we have

https://github.com/OmarManzoor/scikit-learn/blob/5122ff8bbbe0f1d4a3ba7da62d73de69968f46c5/sklearn/utils/extmath.py#L212

But I can't find `toarray` in [the array API standard](https://data-apis.org/array-api/latest/search.html?q=toarray). So what am I missing?

Reply:
That part of the code only runs for sparse arrays. So it isn't linked to the array api.